pkg/activator/net/throttler.go

```diff
@@ -559,36 +618,10 @@ func inferIndex(eps []string, ipAddress string) int {
 	return idx
 }
 
-func (t *Throttler) updateAllThrottlerCapacity() {
-	t.revisionThrottlersMutex.RLock()
-	defer t.revisionThrottlersMutex.RUnlock()
-
-	for _, rt := range t.revisionThrottlers {
-		rt.updateCapacity(t, rt.backendCount)
-	}
-}
-
-func (t *Throttler) activatorEndpointsUpdated(newObj interface{}) {
+func (t *Throttler) publicEndspointsUpdated(newObj interface{}) {
 	endpoints := newObj.(*corev1.Endpoints)
-
-	// We want to pass sorted list, so that we get _some_ stability in the results.
-	epSet, _ := endpointsToDests(endpoints, networking.ServicePortNameHTTP1)
-	eps := epSet.List()
-	t.logger.Debugf("All Activator IPS: %v, my IP: %s", eps, t.ipAddress)
-	idx := inferIndex(eps, t.ipAddress)
-	activatorCount := len(eps)
-	t.logger.Infof("Got %d ready activator endpoints, our position is: %d", activatorCount, idx)
-	atomic.StoreInt32(&t.numActivators, int32(activatorCount))
-	atomic.StoreInt32(&t.activatorIndex, int32(idx))
-	t.updateAllThrottlerCapacity()
-}
-
-func (t *Throttler) index() int {
-	return int(atomic.LoadInt32(&t.activatorIndex))
-}
-
-func (t *Throttler) activatorCount() int {
-	return int(atomic.LoadInt32(&t.numActivators))
+	t.logger.Infof("Public Endpoints %s updated", endpoints.Name)
+	t.epsUpdateCh <- endpoints
```

Comment:
Is this necessary? Couldn't we call into `handlePubEpsUpdate` directly? Or is this "easier" for lifecycle management in tests?

Reply:
this ensures that only one update is processed by the revision throttler at a time so, e.g. updates from backend revision manager and from public service don't step on each other's toes — thus reducing requirement for locking.
And this is quite possible, since adding a pod (trigger backend revision watcher) might trigger the number of activators in the public service (subsetting).